### ISSUE TITLE
sharding 4.0.0-RC2+seata 0.9.0, error :Caused by: java.sql.SQLFeatureNotSupportedException: ParameterMetaData

### ISSUE THREAD

**Issue by dengjin001** [apache/incubator-shardingsphere#3551]
## Question
pom.xml:

```xml
<dependency>
            <groupId>org.apache.shardingsphere</groupId>
            <artifactId>sharding-jdbc-spring-boot-starter</artifactId>
        </dependency>
        <dependency>
            <groupId>org.apache.shardingsphere</groupId>
            <artifactId>sharding-transaction-base-seata-at</artifactId>
            <version>4.0.0-RC2</version>
        </dependency>
        <dependency>
            <groupId>io.seata</groupId>
            <artifactId>seata-all</artifactId>
            <version>0.9.0</version>
        </dependency>
```

seataconfig:

```java
@Bean
    public DataSourceProxy dataSourceProxy(DataSource dataSource)  {
        return new DataSourceProxy(dataSource);
    }
    @Bean
    public SqlSessionFactory sqlSessionFactory(DataSourceProxy dataSourceProxy) throws Exception {
        MybatisSqlSessionFactoryBean factoryBean = new MybatisSqlSessionFactoryBean();
        factoryBean.setDataSource(dataSourceProxy);
        return factoryBean.getObject();
    }
```

error message :

```log
Caused by: java.sql.SQLFeatureNotSupportedException: ParameterMetaData
	at org.apache.shardingsphere.shardingjdbc.jdbc.unsupported.AbstractUnsupportedOperationPreparedStatement.getParameterMetaData(AbstractUnsupportedOperationPreparedStatement.java:52)
	at io.seata.rm.datasource.AbstractPreparedStatementProxy.initParameterHolder(AbstractPreparedStatementProxy.java:56)
	at io.seata.rm.datasource.AbstractPreparedStatementProxy.<init>(AbstractPreparedStatementProxy.java:74)
	at io.seata.rm.datasource.PreparedStatementProxy.<init>(PreparedStatementProxy.java:49)
	at io.seata.rm.datasource.AbstractConnectionProxy.prepareStatement(AbstractConnectionProxy.java:118)
	at org.apache.ibatis.executor.statement.PreparedStatementHandler.instantiateStatement(PreparedStatementHandler.java:86)
	at org.apache.ibatis.executor.statement.BaseStatementHandler.prepare(BaseStatementHandler.java:88)
	at org.apache.ibatis.executor.statement.RoutingStatementHandler.prepare(RoutingStatementHandler.java:59)
	at com.baomidou.mybatisplus.core.executor.MybatisSimpleExecutor.prepareStatement(MybatisSimpleExecutor.java:94)
	at com.baomidou.mybatisplus.core.executor.MybatisSimpleExecutor.doUpdate(MybatisSimpleExecutor.java:53)
	at org.apache.ibatis.executor.BaseExecutor.update(BaseExecutor.java:117)
	at org.apache.ibatis.executor.CachingExecutor.update(CachingExecutor.java:76)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.update(DefaultSqlSession.java:197)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.insert(DefaultSqlSession.java:184)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:426)
```

**Comment by cherrylzhao** [comment]
@dengjin001 you should not instance DataSourceProxy by yourself when using `sharding-transaction-base-seata-at`. we have integrated it inside ShardingDataSource

**Comment by cherrylzhao** [comment]
fixed it at #4991, if you want to use original Seata microservice, `sharding-transaction-base-seata-at` module should not be introduced.
